implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java

```diff
@@ -512,6 +545,198 @@ private static SSLContext contextFromKeyAndCerts(final Configuration config) {
     }
   }
   
+  /**
+   * Attempts to load the key from a combination of private key file, 
+   * certificate file(s) and Certificate Authority pem.
+   * @param config A non-null config.
+   * @return An instantiated context or null if something went wrong. The
+   * exceptions will be logged.
+   */
+  private static SSLContext contextFromSecrets(final Configuration config) {
+    final Object cert_secret = config.getSecretObject(
+        config.getString(TLS_SECRET_CERT_KEY));
+    if (cert_secret == null) {
+      throw new IllegalStateException("The secret object cannot be null: " 
+          + config.getString(TLS_SECRET_CERT_KEY));
+    }
+    
+    RSAPrivateKey key = null;
+    final Object key_secret = config.getSecretObject(
+        config.getString(TLS_SECRET_KEY_KEY));
+    try {
+      if (key_secret != null) {
+        if (key_secret instanceof RSAPrivateKey) {
+          key = (RSAPrivateKey) key_secret;
+        } else if (key_secret instanceof String || key_secret instanceof byte[]) {
+          final String raw_key;
+          if (key_secret instanceof String) {
+            raw_key = (String) key_secret;
+          } else {
+            raw_key = new String((byte[]) key_secret, Const.UTF8_CHARSET);
```

Comment:
Is this known to always be UTF-8-encoded?

Reply:
No it's an assumption for now until I can get it configured in the secrets provider.